### PR TITLE
Add gitlab access

### PR DESCRIPTION
**Pull request by Cereal84**
Add GitlabSSO class in order to use GitLab as SSO.
It requires some scopes like 'read_user', 'openid' and 'profile' to retrieves information like username and avatar_url.

### CODE REVIEW

File: examples/gitlab.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+@app.get("/auth/callback")
+async def auth_callback(request: Request):
+    """Verify login"""
+    user = await sso.verify_and_process(request)
+    return user
+
+
+if __name__ == "__main__":
+    uvicorn.run(app="examples.github:app", host="127.0.0.1", port=5000)
```
Comment (tomasvotava) [inline examples/gitlab.py:37]:
```suggestion
    uvicorn.run(app="examples.gitlab:app", host="127.0.0.1", port=5000)
```

**Comment by tomasvotava** [comment]
Hi @Cereal84, thanks so much for the contribution and sorry for taking too long. The PR seems alright to me, there only seems to be `github` in the module path as seen in my review.

File: examples/gitlab.py (hunk 1 above)
Comment (Cereal84) [reply examples/gitlab.py:37]: You're right :)

**Comment by Cereal84** [comment]
> Hi @Cereal84, thanks so much for the contribution and sorry for taking too long. The PR seems alright to me, there only seems to be `github` in the module path as seen in my review. 

Hi @tomasvotava, don't worry about the time :).
I've fixed the example.

**Review by tomasvotava — APPROVED** [review]

**Pull request merged**